volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -26,43 +26,89 @@
 
 
 class module(generic.GenericIntelProcess):
+    def _get_mod_mem_type(self):
+        """Attempt to get the mod_mem_type enum once to allow repeated access from other functions"""
+        if not self.has_member("mod_mem_type"):
+            try:
+                vmlinux = linux.LinuxUtilities.get_module_from_volobj_type(
+                    self._context, self
+                )
+                # mod_mem_type and module_memory were added in kernel 6.4 which replaces
+                # module_layout for storing the information around core_layout etc.
+                # see commit ac3b43283923440900b4f36ca5f9f0b1ca43b70e for more information
+                self.mod_mem_type = vmlinux.get_enumeration("mod_mem_type").choices
+            except exceptions.SymbolError:
+                vollog.debug(
+                    f"Unable to find mod_mem_type enum. This is expected on kernels <6.4 but may cause issues with later kernels"
+                )
+                self.mod_mem_type = None
+
     def get_module_base(self):
-        if self.has_member("core_layout"):
-            return self.core_layout.base
+        self._get_mod_mem_type()
+        if self.mod_mem_type:
+            return self.mem[self.mod_mem_type["MOD_TEXT"]].base
```

Comment:
This is an Enum, so we should be able to ensure these values are always present.  The question is *just in case* do we trust the enum values never to be renamed/removed, do we use a get with a default we choose or do we put in a try/catch for a lookup error (and in both those cases what default do we return)?  5:S

Reply:
Another good idea. At first I wanted to use an UnparsableValue from the renderers - but seeing that before it would raise an attribute error if it couldn't find the right values I thought that made more sense? It feels like if the kernel changes the enum enough that the parts we want are not there - than we probably need to update something rather than just a default value? What do you think? It feels like raising the attribute error would highlight exactly where the problem is.